scripts/src/main/resources/scripts/command/ionic

```diff
@@ -10,10 +10,11 @@ then
   exit
 fi
 
-if [ -n "${DEVON_IDE_TRACE}" ]; then set -vx; fi
-TOOL_VERSION_COMMAND="ionic -v"
 # shellcheck source=scripts/functions
 source "$(dirname "${0}")"/../functions
+TOOL_VERSION_COMMAND="ionic -v"
+# shellcheck source=scripts/commandlet-cli
+source "$(dirname "${0}")"/../commandlet-cli
 
 # $1: optional setup
 function doSetup() {
```

Comment:
During the setup of npm a 'MODULE_NOT_FOUND' blocks the  installation of ionic. 
To be more precise, I've been getting the following error after the installation of node seemed
to be complete:


`node:internal/modules/cjs/loader:998
  throw err;
  ^

Error: Cannot find module '/home/dr/projects/PR934/software/node/bin/node_modules/npm/bin/npm-cli.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:995:15)
    at Module._load (node:internal/modules/cjs/loader:841:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:82:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v19.0.0
Could not determine Node.js install directory

******** ATTENTION ********
Failed to install ionic-cli ('/home/dr/projects/PR934/software/node/npm')
We are sorry for the inconvenience. Please check the above errors, resolve them and try again.
Exit code was 1
`


This error could  be unrelated to the changes of this PR,
but seems to be worth mentioning.

Reply:
It seems as if '/home/dr/projects/PR934/software/node/bin/node_modules/npm/bin/npm-cli.js' would be just the wrong location, as the file 'npm-cli.js' exists, but with one bin directory less in the path. So the proper path in my case is '/home/dr/projects/PR934/software/node/bin/node_modules/npm/bin/npm-cli.js'.

Reply:
> So the proper path in my case is '/home/dr/projects/PR934/software/node/bin/node_modules/npm/bin/npm-cli.js'.

Did you mean '/home/dr/projects/PR934/software/node/bin/node_modules/npm/npm-cli.js'?

Reply:
Full log of this error can be found here: https://github.com/alfeilex/ide/actions/runs/3368418467/jobs/5586938443#step:4:11593

Reply:
No, but I actually took away the wrong bin folder. In my case the npm-cli.js file has the following path
/home/dr/projects/PR934/software/node/node_modules/npm/bin/npm-cli.js .

Reply:
https://github.com/devonfw/ide/blob/11db4ef9862f5f7baa8b89e9f3c14d15bd0380dd/scripts/src/main/resources/scripts/command/ionic#L29-L33

It seems that the errors occurs because the `npm_command` uses the path from line 29, but it must be the path of line 32. Probably the condition in line 30 needs to be changed to (e.g. `! doIsWindows`) or something like that

Reply:
@alfeilex that's just the code part that needed to be fixed :+1: 
I tried it out with  `[ -x "${npm_command}" ]` instead of `[ ! -x "${npm_command}" ]` and the setup of ionic works now. 
If macs needs the same path, then `! doIsWindows`  seems to be a better readable way. 
Maybe @hohwille knows more about this?

Reply:
From speaking with @alfeilex now we should most probably use `! doIsWindows` instead, as otherwise the npm_command variable would falsely also be changed for windows.

Reply:
This code pattern is actually bad. We have several npm based commandlets that would need such C&P logic.
Instead we should simply change to `doDevonCommand npm install ...`.
Then only `npm` commandlet should implement the logic how and where to find `npm`.
I can do that.

For the initial comment removing the `!` does not make any sense.
The idea is that we have two options where `npm` can be located. If the first one does not apply because it does not point to an executable we use the second one as fallback.
Maybe you have been facing problems because `npm` was not installed when this decision was made and after install `${npm_command}` was used without doing the check again. We have solved this problem in `aws` with an own `findAws` function that can be called again.

Reply:
I created bug #966 for this. IMHO we need to simply switch the two paths as the second should be used in favor if it exists. However, this has nothing to do we this PR so we can fix it in a separate PR.